compiler/rustc_typeck/src/collect.rs

```diff
@@ -1690,7 +1692,7 @@ pub fn const_evaluatable_predicates_of<'tcx>(
     tcx: TyCtxt<'tcx>,
     def_id: DefId,
     predicates: &ty::GenericPredicates<'tcx>,
-) -> impl Iterator<Item = (ty::Predicate<'tcx>, Span)> {
+) -> Vec<(ty::Predicate<'tcx>, Span)> {
```

Comment:
you can also use `impl Iterator<Item = ...> + ExactSizeIterator` if you want to avoid the `collect`